atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosServerImplementation.java

```diff
@@ -0,0 +1,233 @@
+    }
+
+    @Override
+    public void onStop() {
+        // Nothing to do
+    }
+
+    @Override
+    public void onStartupFailure() {
+        // Nothing
+    }
+
+    @Override
+    public TimeLockServices createInvalidatingTimeLockServices(String client) {
+        TimestampService timestampService = createPaxosBackedTimestampService(client);
+        LockService lockService = AwaitingLeadershipProxy.newProxyInstance(
+                LockService.class,
+                LockServiceImpl::create,
+                leaderElectionService);
+
+        return TimeLockServices.create(timestampService, lockService);
+    }
+
+    private TimestampService createPaxosBackedTimestampService(String client) {
+        paxosResource.addClient(client);
+
+        ExecutorService executor = Executors.newCachedThreadPool(new ThreadFactoryBuilder()
+                .setNameFormat("atlas-consensus-" + client + "-%d")
+                .setDaemon(true)
+                .build());
+
+        List<PaxosAcceptor> acceptors = getNamespacedProxies(
+                remoteServers,
+                client,
+                optionalSecurity,
+                PaxosAcceptor.class);
+        acceptors.add(paxosResource.getPaxosAcceptor(client));
+        List<PaxosLearner> learners = getNamespacedProxies(
+                remoteServers,
+                client,
+                optionalSecurity,
+                PaxosLearner.class);
+        learners.add(paxosResource.getPaxosLearner(client));
+
+        PaxosProposer proposer = PaxosProposerImpl.newProposer(
+                paxosResource.getPaxosLearner(client),
+                ImmutableList.copyOf(acceptors),
+                ImmutableList.copyOf(learners),
+                acceptors.size() / 2 + 1,
+                executor);
+
+        return AwaitingLeadershipProxy.newProxyInstance(
+                TimestampService.class,
+                () -> PersistentTimestampService.create(
+                        new PaxosTimestampBoundStore(
+                                proposer,
+                                paxosResource.getPaxosLearner(client),
+                                ImmutableList.copyOf(acceptors),
+                                ImmutableList.copyOf(learners),
+                                paxosConfiguration.randomWaitBeforeProposalMs())),
+                leaderElectionService);
+    }
+
+    private static Set<String> getRemoteAddresses(TimeLockServerConfiguration configuration) {
+        return Sets.difference(configuration.cluster().servers(),
+                ImmutableSet.of(configuration.cluster().localServer()));
+    }
+
+    private Set<String> getRemotePaths(TimeLockServerConfiguration configuration) {
+        String protocolPrefix = paxosConfiguration.sslConfiguration().isPresent()
+                ? "https://" : "http://";
+        return getRemoteAddresses(configuration).stream()
+                .map(address -> protocolPrefix + address)
```

Comment:
Yes. `ClusterConfiguration::checkHostPortString` verifies that the server addresses are of the form `host:port` (there's also a test that checks this specific case, that `http://foo:bar` gets rejected).